launcher/minecraft/mod/ModFolderModel.cpp

```diff
@@ -124,27 +134,28 @@ void ModFolderModel::finishUpdate()
     QSet<QString> newSet(newList.begin(), newList.end());
 #else
     QSet<QString> currentSet = modsIndex.keys().toSet();
-    auto & newMods = m_update->mods;
+    auto& newMods = m_update->mods;
     QSet<QString> newSet = newMods.keys().toSet();
 #endif
 
     // see if the kept mods changed in some way
     {
         QSet<QString> kept = currentSet;
         kept.intersect(newSet);
-        for(auto & keptMod: kept) {
-            auto & newMod = newMods[keptMod];
+        for(auto& keptMod : kept) {
+            auto* newMod = newMods[keptMod];
```

Comment:
LeakBGone : Not forcing pointer here, to allow smart pointers
```suggestion
            auto newMod = newMods[keptMod];
```